homeassistant/components/srp_energy/__init__.py

```diff
@@ -0,0 +1,53 @@
+"""The SRP Energy integration."""
+import logging
+
+from srpenergy.client import SrpEnergyClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ID, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+PLATFORMS = ["sensor"]
+
+
+async def async_setup(hass, config):
+    """Old way of setting up the srp_energy component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up the SRP Energy component from a config entry."""
+    # Store an SrpEnergyClient object for your srp_energy to access
+    account_id = entry.data.get(CONF_ID)
+    username = entry.data.get(CONF_USERNAME)
+    password = entry.data.get(CONF_PASSWORD)
+    hass.data[DOMAIN] = None
+
+    try:
+        srp_energy_client = SrpEnergyClient(account_id, username, password)
```

Comment:
It is poor practice to do IO when initialising a class

Reply:
I can go back and remove the IO in the SrpEnergyClient. As for now, is there a better way given that's how the client behaves?

Reply:
Sure, just so you are aware